fv3core/stencils/xtp_u.py

```diff
@@ -123,37 +122,46 @@ def _compute_stencil(
         flux = _get_flux(u, courant, rdx, bl, br)
 
 
-def compute(c: FloatField, u: FloatField, flux: FloatField):
-    """
-    Compute flux of kinetic energy in x-dir.
-
-    Args:
-        c (in): Courant number in flux form
-        u (in): x-dir wind on D-grid
-        flux (out): Flux of kinetic energy
-    """
-    grid = spec.grid
-    iord = spec.namelist.hord_mt
-    if iord not in (5, 6, 7, 8):
-        raise NotImplementedError(
-            "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+class XTP_U:
+    def __init__(self, namelist):
+        iord = spec.namelist.hord_mt
+        if iord not in (5, 6, 7, 8):
+            raise NotImplementedError(
+                "Currently xtp_v is only supported for hord_mt == 5,6,7,8"
+            )
+        self.grid = spec.grid
```

Comment:
Not necessary here or for the other first-round object-oriented PRs, but in the long term when we work to pass only the required namelist flags, I'd also like to pass only the required grid variables. This form makes it way easier to do that, since we can grep for `self.grid`!

Reply:
Yeah we just discussed for awhile, I still think it's nice on the user to have the available objects be categorized very clearly where they are used, but you and Oliver pointed out we should need an entire grid object for every module, and fortran actually does also explicitly call out which grid variables are used in each modules. So can move that direction.